vm/rust/src/lib.rs

```diff
@@ -6,36 +6,55 @@ extern crate lazy_static;
 
 use crate::juno_state_reader::{ptr_to_felt, JunoStateReader};
 use std::{
-    collections::HashMap, ffi::{c_char, c_longlong, c_uchar, c_ulonglong, c_void, CStr, CString}, num::NonZeroU128, slice, sync::Arc
+    collections::HashMap,
+    ffi::{c_char, c_longlong, c_uchar, c_ulonglong, c_void, CStr, CString},
+    num::NonZeroU128,
+    slice,
+    sync::Arc,
 };
 
+use blockifier::abi::constants::STORED_BLOCK_HASH_BUFFER;
+use blockifier::blockifier::block::{
+    pre_process_block, BlockInfo as BlockifierBlockInfo, BlockNumberHashPair, GasPrices,
+};
+use blockifier::bouncer::BouncerConfig;
+use blockifier::fee::{fee_utils, gas_usage};
+use blockifier::transaction::objects::GasVector;
 use blockifier::{
-    block::{pre_process_block, BlockInfo as BlockifierBlockInfo, BlockNumberHashPair, GasPrices}, context::{BlockContext, ChainInfo, FeeTokenAddresses, TransactionContext}, execution::{
+    context::{BlockContext, ChainInfo, FeeTokenAddresses, TransactionContext},
+    execution::{
         contract_class::ClassInfo,
         entry_point::{CallEntryPoint, CallType, EntryPointExecutionContext},
-    }, fee::fee_utils::calculate_tx_fee, state::{cached_state::{CachedState, GlobalContractCache}, state_api::State}, transaction::{
+    },
+    state::{cached_state::CachedState, state_api::State},
+    transaction::{
         errors::TransactionExecutionError::{
-            ContractConstructorExecutionFailed,
-            ExecutionError,
-            ValidateTransactionError,
-        }, objects::{DeprecatedTransactionInfo, HasRelatedFeeType, TransactionInfo}, transaction_execution::Transaction, transactions::ExecutableTransaction
-    }, versioned_constants::VersionedConstants
+            ContractConstructorExecutionFailed, ExecutionError, ValidateTransactionError,
+        },
+        objects::{DeprecatedTransactionInfo, HasRelatedFeeType, TransactionInfo},
+        transaction_execution::Transaction,
+        transactions::ExecutableTransaction,
+    },
+    versioned_constants::VersionedConstants,
 };
 use cairo_vm::vm::runners::cairo_runner::ExecutionResources;
 use juno_state_reader::{class_info_from_json_str, felt_to_byte_array};
 use serde::Deserialize;
-use starknet_api::{block::BlockHash, core::PatriciaKey, transaction::{Calldata, Transaction as StarknetApiTransaction, TransactionHash}};
 use starknet_api::{
-    deprecated_contract_class::EntryPointType,
-    hash::StarkFelt,
-    transaction::Fee,
+    block::BlockHash,
+    core::PatriciaKey,
+    transaction::{Calldata, Transaction as StarknetApiTransaction, TransactionHash},
 };
 use starknet_api::{
     core::{ChainId, ClassHash, ContractAddress, EntryPointSelector},
     hash::StarkHash,
 };
+use starknet_api::{deprecated_contract_class::EntryPointType, transaction::Fee};
```

Comment:
Any reason why this could not be joined with line 43 above?

Reply:
Nope, merged it